### PR TITLE
(PC-8207) ios booking impossible send email

### PR DESCRIPTION
**Pull request by emmajmn**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-8207

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [ ] Written **documentation on Notion** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/bookOffer/services/useNotifyWebappLinkSent.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { useMutation } from 'react-query'
+
+import { api } from 'api/api'
+
+interface NotifyWebappLinkSentInterface {
+  offerId?: number | undefined
+}
+
+export const useNotifyWebappLinkSent = ({ offerId }: NotifyWebappLinkSentInterface) => {
+  return useMutation(async () => {
```
Comment (marionnousvalentin) [inline src/features/bookOffer/services/useNotifyWebappLinkSent.ts:10]: 🏗️ ça serait plus court de faire 
if(isNil(offerId)) return non ?

File: src/features/bookOffer/services/useNotifyWebappLinkSent.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    const callSendWebappLinkByEmail =
+      offerId && api.postnativev1sendOfferWebappLinkByEmailofferId(offerId)
+    const callSendWebappLinkByPushNotif =
+      offerId && api.postnativev1sendOfferLinkByPushofferId(offerId)
+    return Promise.all([await callSendWebappLinkByEmail, await callSendWebappLinkByPushNotif])
```
Comment (marionnousvalentin) [inline src/features/bookOffer/services/useNotifyWebappLinkSent.ts:15]: ❓ ici c'est bien ce que l'on veut dans lancer les deux calls en parallèle indépendamnt de de ail si l'un des deux échoue ?

**Review by marionnousvalentin — APPROVED** [review]

File: src/features/bookOffer/services/useNotifyWebappLinkSent.ts (hunk 2 above)
Comment (emmajmn) [reply src/features/bookOffer/services/useNotifyWebappLinkSent.ts:15]: yes :)

File: src/features/bookOffer/services/useNotifyWebappLinkSent.ts (hunk 1 above)
Comment (emmajmn) [reply src/features/bookOffer/services/useNotifyWebappLinkSent.ts:10]: yes je peux changer

**Pull request merged**